src/sherlock/nginx_tailer.py

```diff
@@ -33,7 +33,18 @@ async def tail_nginx_log() -> None:
             )
 
             while True:
-                line = await resp.content.readline()
+                # apparently readline() times out after six minutes
+                # ...so you'd think you could just catch that error and
+                # retry, but then you get intermittent pod-not-ready
+                # errors
+                # ...so let's try a slightly smaller controlled timeout
+                # and see if that helps any.
+                #
+                # Spoiler: it did.
+                try:
+                    line = await asyncio.wait_for(resp.content.readline(), 300)
+                except asyncio.TimeoutError:
+                    continue
```

Comment:
I don't think this will work... the continue will rerun line 35 and we'll just keep trying to read from probably a dead response?  It seems like that should maybe be a break so that it will make a new query up on line 21.  But I think that's what it does without this, since we're already in a try.

What error were you hitting about the pod not ready?  I'm very confused about this, what pod wasn't ready?

Reply:
I have seen the timeout happen but when it does it just normally picks up from where it left off, so I'm not sure if you're seeing something else?